ballerina/http_response.bal

```diff
@@ -541,12 +541,12 @@ public class Response {
     }
 
     isolated function buildStatusCodeResponse(typedesc<anydata>? payloadType, typedesc<StatusCodeResponse> statusCodeResType,
-        boolean requireValidation, Status status, map<string|int|boolean|string[]|int[]|boolean[]> headers, string? mediaType)
-            returns StatusCodeResponse|ClientError {
+        boolean requireValidation, Status status, map<string|int|boolean|string[]|int[]|boolean[]> headers, string? mediaType,
+        boolean fromDefaultStatusCodeMapping) returns StatusCodeResponse|ClientError {
```

Comment:
This method is taking around 7 parameters, which reduce the readability significantly. Couldn't we refactor this method to something like this ?

```ballerina
type StatsCodeRespContext record {|
    typedesc<anydata> payloadType?;
    typedesc<StatusCodeResponse> statusCodeResType;
    boolean requireValidation; 
    Status status;
    map<string|int|boolean|string[]|int[]|boolean[]> headers;
    string mediaType?;
    boolean fromDefaultStatusCodeMapping;
|};

isolated function buildStatusCodeResponse(StatsCodeRespContext context) returns StatusCodeResponse|ClientError {
    // function logic here         
}
```

Reply:
This is called from the java native function and used for internal purpose. Having a record means we need to build it from java native side and call this function, which adds an additional overhead for the function call

Reply:
What if we pass this as an included record param ?
```ballerina
isolated function buildStatusCodeResponse(*StatsCodeRespContext context) returns StatusCodeResponse|ClientError;
```

Would this fix it ?

Reply:
For included parameter, we have to pass it as a BMap from the Java side. So this is same as having a record parameter